eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/ManagementTypeFactory.cs

```diff
@@ -31,5 +42,47 @@ public class ManagementTypeFactory : AzureTypeFactory
             var transformedClient = InputClientTransformer.TransformInputClient(inputClient);
             return transformedClient is null ? null : base.CreateClientCore(transformedClient);
         }
+
+        /// <inheritdoc/>
+        protected override CSharpType? CreateCSharpTypeCore(InputType inputType)
+        {
+            if (inputType is InputModelType model && KnownManagementTypes.TryGetManagementType(model.CrossLanguageDefinitionId, out var replacedType))
+            {
+                return replacedType;
+            }
+            return base.CreateCSharpTypeCore(inputType);
+        }
+
+        /// <inheritdoc/>
+        protected override ModelProvider? CreateModelCore(InputModelType model)
+        {
+            if (KnownManagementTypes.TryGetManagementType(model.CrossLanguageDefinitionId, out var replacedType))
+            {
+                return new InheritableSystemObjectModelProvider(replacedType.FrameworkType, model);
+            }
+            return base.CreateModelCore(model);
+        }
+
+        /// <inheritdoc/>
+        public override MethodBodyStatement SerializeJsonValue(Type valueType, ValueExpression value, ScopedApi<Utf8JsonWriter> utf8JsonWriter, ScopedApi<ModelReaderWriterOptions> mrwOptionsParameter, SerializationFormat serializationFormat)
+        {
+            if (KnownManagementTypes.IsKnownManagementType(valueType))
+            {
+                return Static(typeof(JsonSerializer)).Invoke(nameof(JsonSerializer.Serialize), [value]).Terminate();
```

Comment:
Should this be calling the `ModelReaderWriter`?

Reply:
The original behavior is using JsonSerializer, will leave it as it is.